src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -61,6 +61,9 @@ def __init__(self, config):
         self.topicAMQ = getattr(config.AgentStatusWatcher, "topicAMQ", None)
         self.hostPortAMQ = getattr(config.AgentStatusWatcher, "hostPortAMQ", [('cms-mb.cern.ch', 61313)])
 
+        # WorkQueue replication
+        self.workqueueSelector = getattr(config.AgentStatusWatcher, "workqueueSelector", {})
```

Comment:
I am inclined to think that we should not import the selector from the configuration but keep it in the code here. I do not expect the selector to be different across wmagent instances, and it was part of "code" and not "configuration" before, so i do not see any reason to move the selector logic away from where it will be used.

Am i missing anything?

Reply:
Please don't mix Erlang code with Python, keep Erlang code apart since if you merge it here you'll be forced to change Python code every time you change a selector. While if you keep it separate you only may need to change it once and restart the Python code. From my experience, e.g. DBS using embedded SQL statements, it is much harder to maintain the code with embedded snippets than otherwise. I prefer to keep such stuff separate, e.g. for that separate area exists which can keep JS, images, templates, erlang, etc.

Reply:
I provided my feedback for consideration. Given that the content is in JSON format, it might be tempting to incorporate it directly into Python code. However, doing so could blur the distinction, as Python does not differentiate between JSON and Python dictionaries, treating both the same way. This could create ambiguity—whether it's a standalone JSON structure or a Python map or object.

If maintaining clear separation between models (i.e., distinguishing what belongs to Python versus Erlang or other systems) is a priority, then I believe such separation is beneficial. However, I'll leave the final decision up to you. If you will find putting JSON in place I suggest to put appropriate comment that it is JSON (not python map) and where it belongs (Erlang).